Invoke onshippingcontactselected when "postalAddress" is passed in "requiredShippingContactFields"

mariovalney opened the pull request:
Fixes #8

mariovalney commented:
I'm not sure why the df4f9432d708e8df4084ce7906fa5ff721571fc8 commit was necessary.

I checked editorconfig and spaces. Any clues to recreate the PR without this?

martincostello commented:
Looks like this repo doesn't actually have ~an `.editorconfig` file in it, nor~ a `.gitattributes`. I'll look to add an appropriate one ~of each~, then if you rebase once I've done that _hopefully_ your change should apply without all the whitespace changes.

martincostello commented:
That's done. I had it do something weird just now with changes, but then it stopped doing it in Visual Studio Code. Now quite sure what's happening.

mariovalney committed:
Feat: added timeout to onshippingcontactselected

Docs: https://developer.apple.com/documentation/apple_pay_on_the_web/applepaysession/1778009-onshippingcontactselected

mariovalney committed:
Feat: checks for 'requiredShippingContactFields'

On payment request to call the 'onshippingcontactselected' if needed.

mariovalney commented:
Rebased! Thanks :)

martincostello commented on ApplePaySession.js:
```diff
@@ -337,13 +346,15 @@
                     };
 
-                    session.onshippingcontactselected(applePayShippingContactSelectedEvent);
+                    self.onShippingContactSelectedWithTimeout(session, applePayShippingContactSelectedEvent)
```
```suggestion
                    self.onShippingContactSelectedWithTimeout(session, applePayShippingContactSelectedEvent);
```

mariovalney committed:
Update ApplePaySession.js

Co-authored-by: Martin Costello <martin@martincostello.com>

martincostello commented:
Thanks again 👍
